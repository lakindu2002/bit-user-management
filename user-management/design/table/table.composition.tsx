import React from 'react';
import { Table } from './table';

const tableHeaders = [
  { key: 'name', label: 'Name' },
  { key: 'age', label: 'Age' },
  { key: 'location', label: 'Location' },
];

const tableItems: Record<string, string>[] = [
  { name: 'John Doe', age: '30', location: 'New York' },
  { name: 'Jane Smith', age: '25', location: 'Los Angeles' },
  { name: 'Michael Johnson', age: '28', location: 'Chicago' },
];

export const BasicTable = () => {
  return (
    <div>
      <h1>Dynamic Table Example</h1>
      <Table headers={tableHeaders} items={tableItems} />
    </div>
  );
}
