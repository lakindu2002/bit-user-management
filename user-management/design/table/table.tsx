import React from 'react';

export type TableProps = {
  headers: { key: string; label: string }[];
  items: Record<string, string>[];
};

const tableHeaderStyle: React.CSSProperties = {
  backgroundColor: '#f2f2f2',
  padding: '8px',
  textAlign: 'left',
};

const tableCellStyle: React.CSSProperties = {
  border: '1px solid #dddddd',
  padding: '8px',
};

export function Table({ headers, items }: TableProps) {
  return (
    <table style={{ borderCollapse: 'collapse', width: '100%' }}>
      <thead>
        <tr>
          {headers.map(({ key, label }, index) => (
            <th key={key} style={tableHeaderStyle}>
              {label}
            </th>
          ))}
        </tr>
      </thead>
      <tbody>
        {items.map((item, rowIndex) => (
          <tr key={rowIndex}>
            {headers.map(({ key }) => (
              <td key={key} style={tableCellStyle}>
                {item[key]}
              </td>
            ))}
          </tr>
        ))}
      </tbody>
    </table>
  );
}
