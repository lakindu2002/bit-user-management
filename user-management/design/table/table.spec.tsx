import React from 'react';
import { render } from '@testing-library/react';
import { Table } from './table';

describe('Table component', () => {
  const headers = [
    { key: 'name', label: 'Name' },
    { key: 'age', label: 'Age' },
    { key: 'location', label: 'Location' },
  ];

  const items = [
    { name: 'John Doe', age: '30', location: 'New York' },
    { name: 'Jane Smith', age: '25', location: 'Los Angeles' },
  ];

  it('renders headers and items correctly', () => {
    const { getByText } = render(<Table headers={headers} items={items} />);

    headers.forEach(header => {
      expect(getByText(header.label)).toBeInTheDocument();
    });

    items.forEach(item => {
      Object.values(item).forEach(value => {
        expect(getByText(value)).toBeInTheDocument();
      });
    });
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Table headers={headers} items={items} />);
    expect(asFragment()).toMatchSnapshot();
  });
});