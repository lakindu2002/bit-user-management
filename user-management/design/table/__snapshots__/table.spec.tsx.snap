// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component matches snapshot 1`] = `
<DocumentFragment>
  <table
    style="border-collapse: collapse; width: 100%;"
  >
    <thead>
      <tr>
        <th
          style="background-color: rgb(242, 242, 242); padding: 8px; text-align: left;"
        >
          Name
        </th>
        <th
          style="background-color: rgb(242, 242, 242); padding: 8px; text-align: left;"
        >
          Age
        </th>
        <th
          style="background-color: rgb(242, 242, 242); padding: 8px; text-align: left;"
        >
          Location
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          style="border: 1px solid #dddddd; padding: 8px;"
        >
          John Doe
        </td>
        <td
          style="border: 1px solid #dddddd; padding: 8px;"
        >
          30
        </td>
        <td
          style="border: 1px solid #dddddd; padding: 8px;"
        >
          New York
        </td>
      </tr>
      <tr>
        <td
          style="border: 1px solid #dddddd; padding: 8px;"
        >
          Jane Smith
        </td>
        <td
          style="border: 1px solid #dddddd; padding: 8px;"
        >
          25
        </td>
        <td
          style="border: 1px solid #dddddd; padding: 8px;"
        >
          Los Angeles
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
